src/GraphQL.php

```diff
@@ -180,8 +182,12 @@ public function executeQuery(
         // This allows tracking the time for batched queries independently.
         $this->prepSchema();
 
+        if (is_string($query)) {
+            $query = Parser::parse($query);
```

Comment:
I'm personally thinking it makes sense that the event (and the EndExection) doesn't fire because technically you won't start executing the query.

Maybe (but not sure if that makes it easier) events firing around the query parsing could also be interesting, although I must admit I'm not a 100% sure what the parser does, if it's just syntax or also resolving code and types (because if it does it might be interesting to know how long that part takes)?

Reply:
Parsing is pretty simple, it does not even consider the schema, just syntax. When in doubt, looking at the spec can sometimes be a good idea. In their definition of *Execution*, the query string is already parsed: http://spec.graphql.org/draft/#sec-Execution

So I think we can replicate the error handling around syntax errors and include the parsed query in the event and not fire it when we don't even enter execution.

Reply:
When in doubt RTFM I guess, smart!

This sounds pretty logical to me too 👍